src/App.tsx

```diff
@@ -15,49 +20,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+interface ReorderOptions {
+  sortType: SortType,
+  isReversed: boolean,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const goodsToShow = [...goods];
+
+  goodsToShow.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    goodsToShow.reverse();
+  }
+
+  return goodsToShow;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const sortByLength = () => setSortType(SortType.LENGTH);
+
+  const sortByAlphabet = () => setSortType(SortType.ALPHABET);
+
+  const reverse = () => setIsReversed((state) => !state);
```

Comment:
Better to write it explicitly
```suggestion
  const toggleReverse = () => setIsReversed((state) => !state);
```